crates/core/src/db/messages/commit.rs

```diff
@@ -91,21 +131,36 @@ impl Commit {
         count
     }
 
-    pub fn encode(&self, bytes: &mut Vec<u8>) {
-        bytes.reserve(self.encoded_len());
```

Comment:
It's a little unfortunate that we're losing this optimization, but maybe we can improve things by not using a Vec at all but a buffer pool or something in the future.

Reply:
Good point. I meant to make sure the caller takes care of this, will add.

Reply:
e05f58eedb0466de78537f295594a386fa571420